crates/vault-registry/src/lib.rs

```diff
@@ -651,6 +670,12 @@ pub mod pallet {
     pub(super) type TotalUserVaultCollateral<T: Config> =
         StorageMap<_, Blake2_128Concat, DefaultVaultCurrencyPair<T>, BalanceOf<T>, ValueQuery>;
 
+    /// Tuple of (semver_version, sha256_checksum) indicating the latest vault client release.
+    /// `semver_version`: The element at index zero is the major semver version.
+    #[pallet::storage]
+    #[pallet::getter(fn latest_client_release)]
+    pub(super) type LatestClientRelease<T: Config> = StorageValue<_, ClientRelease, ValueQuery>;
```

Comment:
Rename to `CurrentClientRelease` or something and add `PendingClientRelease` to be called alongside `authorize_upgrade`. Set `CurrentClientRelease` to `PendingClientRelease` in the `on_runtime_upgrade` hook.